packages/tokamak/contracts-bedrock/scripts/start-deploy.sh

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env bash
+
+configFilePath=
+environement=
+chainID=
+deployResultFile=
+
+OPTSTRING=":c:e:"
+
+projectRoot=`pwd | sed 's%\(.*/tokamak-thanos\)/.*%\1%'`
+currentPWD=$(pwd)
+
+handleScriptInput() {
+  echo "Check script input"
+  while getopts ${OPTSTRING} opt; do
+    case ${opt} in
+      c)
+        configFilePath=$OPTARG
+        ;;
+      e)
+        echo "Has env file"
+        environement=$OPTARG
+        ;;
+      :)
+        echo "Error: Option -$OPTARG requires an argument." >&2
+        exit 1
+        ;;
+      *)
+        echo "Error: Invalid option -$OPTARG" >&2
+        exit 1
+        ;;
+    esac
+  done
+
+  export DEPLOY_CONFIG_PATH=$configFilePath
+  source $environement
+
+  export chainID=$(jq '.l1ChainID' $configFilePath)
+  echo "ChainID: $chainID"
+
+  echo "Project root: $projectRoot"
+  echo "Current dir: $currentPWD"
+  echo "Config file path: $configFilePath"
+  echo "Env file: $environment"
+
+  # checkScriptInput
+  # checkAccount
+}
+
+updateSystem() {
+  echo "Update system"
+  apt update
+  DEBIAN_FRONTEND=noninteractive apt install git curl wget cmake jq build-essential -y
+}
+
+# checkScriptInput() {
+#   echo "Checking input"
+# }
+
+# checkAccount() {
+#   echo "Verifying accounts"
+# }
+
+installDependency() {
+  installGo
+  installJSPackages
+  installFoundry
+}
+
+installGo() {
+  echo "Install Go"
+  export PATH=$PATH:/usr/local/go/bin
+  if ! go version &> /dev/null; then
+    echo "Installing Go..."
+    cd /root
+    wget https://go.dev/dl/go1.21.13.linux-amd64.tar.gz
+    tar xvzf go1.21.13.linux-amd64.tar.gz
+    mv go /usr/local
+    export PATH=$PATH:/usr/local/go/bin
+    echo PATH=$PATH:/usr/local/go/bin >> ~/.bashrc
+    echo "   Installed $(go version)"
+  else
+    echo "   Found $(go version)"
+  fi
+}
+
+installJSPackages() {
+  curl -fsSL https://deb.nodesource.com/setup_18.x | bash -
+  DEBIAN_FRONTEND=noninteractive apt install -y nodejs
+  DEBIAN_FRONTEND=noninteractive apt install -y npm
+  npm install -g pnpm
+  curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | gpg --dearmor | tee /usr/share/keyrings/yarnkey.gpg >/dev/null
+  echo "deb [signed-by=/usr/share/keyrings/yarnkey.gpg] https://dl.yarnpkg.com/debian stable main" | tee /etc/apt/sources.list.d/yarn.list
+  apt update && apt install yarn
+}
+
+installFoundry() {
+  echo "Install Foundry"
+  if ! forge --version &> /dev/null; then
+    curl -L https://foundry.paradigm.xyz | bash
+    mv /root/.foundry/bin/* /usr/bin/
+    source ~/.bashrc
+    foundryup
+    echo "export PATH=/root/.foundry/bin:$PATH" >> /root/.bashrc
+    source /root/.bashrc
+    export PATH=/root/.foundry/bin:$PATH
+    echo $PATH
+    echo "   Installed $(forge --version)"
+  else
+    echo "   Found $(forge --version)"
+  fi
+}
+
+buildSource() {
+  echo "Start buiding source code"
+  source ~/.bashrc
+  cd $projectRoot
+  pnpm install
+  make submodules
```

Comment:
In a dependency installation, Go appears to be installed fine, but Go is not found here. Can you please check?

Reply:
Now I found some error during `make cannon-prestate` and `make op-node` the error log is here:
```
make[1]: *** [Makefile:24: op-program-host] Error 1
make[1]: Leaving directory '/tokamak-thanos/op-program'
make: *** [Makefile:138: op-program] Error 2
make -C ./op-node op-node
make[1]: Entering directory '/tokamak-thanos/op-node'
env GO111MODULE=on GOOS= GOARCH= go build -v -ldflags "-X main.GitCommit=1e6182502150c049ae354370a7d7165cf249cb0d -X main.GitDate=1733470411 -X github.com/ethereum-optimism/optimism/op-node/version.Version=untagged -X github.com/ethereum-optimism/optimism/op-node/version.Meta=" -o ./bin/op-node ./cmd/main.go
runtime/cgo
github.com/tokamak-network/tokamak-thanos/op-service/ioutil
# runtime/cgo
gcc: error: unrecognized command-line option '-m64'
github.com/tokamak-network/tokamak-thanos/op-service/jsonutil
make[1]: *** [Makefile:29: op-node] Error 1
make[1]: Leaving directory '/tokamak-thanos/op-node'
make: *** [Makefile:110: op-node] Error 2
```

However, we don't need to build whole packages to deploy contracts to L1. So we can remove this part if we need much time to resolve the issue.

Reply:
the gcc version in my env:
```
gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/aarch64-linux-gnu/11/lto-wrapper
Target: aarch64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 11.4.0-1ubuntu1~22.04' --with-bugurl=file:///usr/share/doc/gcc-11/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-11 --program-prefix=aarch64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libquadmath --disable-libquadmath-support --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --enable-fix-cortex-a53-843419 --disable-werror --enable-checking=release --build=aarch64-linux-gnu --host=aarch64-linux-gnu --target=aarch64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-serialization=2
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 11.4.0 (Ubuntu 11.4.0-1ubuntu1~22.04) 
```

Reply:
Sorry, we've already had this conversation. Thanks for checking in!